specification/managednetworkfabric/resource-manager/Microsoft.ManagedNetworkFabric/preview/2024-02-15-preview/InternetGatewayRules.json

```diff
@@ -0,0 +1,510 @@
+          "InternetGatewayRules_ListBySubscription_MaximumSet_Gen": {
+            "$ref": "./examples/InternetGatewayRules_ListBySubscription_MaximumSet_Gen.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "InternetGatewayRuleProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "required": [
+        "ruleProperties"
+      ],
+      "allOf": [
+        {
+          "$ref": "./common.json#/definitions/AnnotationResource",
+          "description": "Common properties for annotated type."
+        }
+      ],
+      "properties": {
+        "ruleProperties": {
+          "description": "Rules for the InternetGateways",
+          "$ref": "#/definitions/RuleProperties"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the resource.",
+          "$ref": "./common.json#/definitions/ProvisioningState",
+          "readOnly": true
+        },
+        "internetGatewayIds": {
+          "type": "array",
+          "description": "List of Internet Gateway resource Id.",
+          "items": {
+            "description": "ARM Resource ID of the Internet Gateway.",
+            "type": "string"
+          },
+          "readOnly": true
+        }
+      },
+      "description": "Internet Gateway Rule Properties defines the resource properties."
+    },
+    "InternetGatewayRule": {
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Resource properties.",
+          "$ref": "#/definitions/InternetGatewayRuleProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "description": "The Internet Gateway Rule resource definition."
+    },
+    "InternetGatewayRulePatch": {
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "allOf": [
+        {
+          "description": "Resource tags",
+          "$ref": "./common.json#/definitions/TagsUpdate"
+        }
+      ],
+      "description": "The Internet Gateway Rules patch resource definition."
+    },
+    "InternetGatewayRulesListResult": {
+      "type": "object",
+      "description": "List of Internet Gateway Rules.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of Internet Gateway Rule resources.",
+          "items": {
+            "$ref": "#/definitions/InternetGatewayRule"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Url to follow for getting next page of resources."
+        }
+      }
+    },
+    "RuleProperties": {
+      "type": "object",
+      "description": "Rules for the InternetGateways",
+      "required": [
+        "action",
+        "condition"
```

Comment:
>condition [](http://example.com/codeflow?start=9&length=9)

It seems like action and condition are meaningless without an address list (source or destination). But they cannot be made required. 

If it were modelled as: 
addressList, and addressType (enum: [source, destination'] ) then all could be made required and more complete. 
However, i understand if this already existing model and cannot be changed.

Reply:
After seeing the examples, i m confused. 
It seems like the condition applies on source and/or destination address list? 
If thats the case, the destination and source addresslists can be made required.